test/common/http/codec_impl_fuzz_test.cc

```diff
@@ -297,14 +300,41 @@ class HttpStream : public LinkedObject<HttpStream> {
       break;
     }
     case test::common::http::DirectionalAction::kTrailers: {
+      bool encoded_trailers = false;
+      // Allow trailers for http >= 2 or for http1x only, when they are configured.
       if (state.isLocalOpen() && state.stream_state_ == StreamState::PendingDataOrTrailers) {
-        if (response) {
-          state.response_encoder_->encodeTrailers(
-              fromSanitizedHeaders<TestResponseTrailerMapImpl>(directional_action.trailers()));
+        if (http_protocol_ > Protocol::Http11) {
+          if (response) {
+            state.response_encoder_->encodeTrailers(
+                fromSanitizedHeaders<TestResponseTrailerMapImpl>(directional_action.trailers()));
+          } else {
+            state.request_encoder_->encodeTrailers(
+                fromSanitizedHeaders<TestRequestTrailerMapImpl>(directional_action.trailers()));
+          }
+          encoded_trailers = true;
         } else {
-          state.request_encoder_->encodeTrailers(
-              fromSanitizedHeaders<TestRequestTrailerMapImpl>(directional_action.trailers()));
+          // For HTTP1 we need to see if the given direction supports encoding
+          // trailers.
+          if (response && http1_allow_server_trailers_) {
+            // TODO(kbaichoo): guard to see if can encode trailers by seeing if
+            // content length missing from hdrs so we support trailers.
+            state.response_encoder_->encodeTrailers(
+                fromSanitizedHeaders<TestResponseTrailerMapImpl>(directional_action.trailers()));
+            encoded_trailers = true;
```

Comment:
Just for my understanding: We are encoding trailers on the server side here, but settings this bool to close the client side. Why does this make sense? Wouldn't it be "more natural" to close the server side (remote) here?